packages/integration-tester/src/validate.ts

```diff
@@ -0,0 +1,148 @@
+import { execa } from "execa";
+import { readdir } from "fs/promises";
+import { globby } from "globby";
+import { dirname, relative } from "pathe";
+import pc from "picocolors";
+import type { IntegrationTestSuite } from "./config/types.js";
+import type { TaskRunner } from "./runner.js";
+import { log } from "./utils.js";
+
+// Number of parallel TypeSpec compilations to run
+const COMPILATION_CONCURRENCY = 4;
```

Comment:
```suggestion
const COMPILATION_CONCURRENCY = os.cpus().length;
```

Why not default to number of cores?